components/centraldashboard/app/api.ts

```diff
@@ -17,6 +18,31 @@ export function api(kubeConfig: KubeConfig): express.Router {
           async (_: express.Request, res: express.Response) => {
             res.json(await k8sService.getNamespaces());
           })
+      .get(
+          '/metrics/:type((node|podcpu|podmem))',
+          async (req: express.Request, res: express.Response) => {
+            let interval = Interval.Last15m;
+            if (Interval[req.query.interval] !== undefined) {
+              interval = Number(Interval[req.query.interval]);
+            }
+            if (metricsService) {
```

Comment:
Just a suggestion, to code-reuse:
```js
if (!metricsService) return res.sendStatus(405);
const method = {
    'node': 'getNodeCpuUtilization',
    'podcpu': 'getPodCpuUtilization',
    'podmem': 'getPodMemoryUsage',
}[req.params.type]
return res.json(await metricsService[method](interval));
```

Reply:
Typescript won't permit this sort of syntax since it makes it impossible for the type checker to know the types of the methods involved. While I would agree that this is more succinct in terms of lines of codes, I would contend that it's less obvious what the code is doing especially to a first time reader. I've always tried to apply the axiom from the Zen of Python that *explicit is better than implicit*. Practically speaking, I'd rather type a few more lines of code if it saves me the trouble of inspecting what that code is doing 6 or 12 months from now and most of the Google language style rules advise the same.

I'm definitely open to suggestions though, particularly if you have a set of style principles that you think should apply for scenarios like this.

Reply:
Makes sense, I wasn't sure. I've grown the style after writing a tons of js, and probably why I am really biased to not knowing when ES2016+ esque code gets less readable.

On the other hand, early return and handling is known generally to be a good design pattern:
- [Software Engineering StackExchange](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)
- [Reddit](https://www.reddit.com/r/programming/comments/878zi2/avoid_else_return_early/)
- [HackerNews](https://news.ycombinator.com/item?id=16678209)
- Google tott